src/useradd.c

```diff
@@ -2073,7 +2079,35 @@ static void create_home (void)
 			strcat (path, "/");
 			strcat (path, cp);
 			if (access (path, F_OK) != 0) {
-				if (mkdir (path, 0) != 0) {
+				/* Check if parent directory is BTRFS, fail if requesting
```

Comment:
I have three issues with this (besides the existing formatting ARGH)

1. if the homedirs for some reason are something like /home/group1/group2//user1, and all of it is btrfs, then you will create a new btrfs subvolume for each directory that didn't exist.
2. Like Christian said, it seems like btrfs_check handling is awkward.  It's bikeshedding to a degree, though, and what i was going to suggest ends up with more work in the !btrfs case...
3. This is now way too much code for this simple "create if doesn't exist" loop.  The btrfs creation should be in its own function, and for that matter in the existing code there should just be a "mkdir_or_die()" call.